### ISSUE TITLE
How do I restore a face image from a .npy file?

### ISSUE THREAD

**Issue by tianjialiang12** [yuval-alaluf/restyle-encoder#32]
Just like stylegan encoder, I want to fuse two faces, thanks for your answer.

**Comment by tianjialiang12** [comment]
These are my codes and results. What's wrong with my code? The result is white instead of a face picture.
![QQ截图20210817113949](https://user-images.githubusercontent.com/69072516/129660144-0df0e022-3b1b-4557-a5c0-80d7f7e95d0c.jpg)
![QQ截图20210817113938](https://user-images.githubusercontent.com/69072516/129660147-1dc7e374-a0a0-4ba0-99eb-28d9bef7c2a0.jpg)

**Comment by yuval-alaluf** [comment]
I believe the issue is centered around what you're doing in the postprocessing on `src_img`. Try doing the following instead:
```python
src_img, _ = net.decodr([src_latent], input_is_latent=True, randomize_noise=False)
image = tensor2im(src_img)
```
Also, I am also not sure without trying, but you may not want to perform `unsqueeze(0)` before passing the latent to the decoder so you should verify that this is indeed correct. 
If it helps, here is a function that generates random images using StyleGAN: 
```python
def get_random_image(net):
    with torch.no_grad():
        vecs_to_inject = np.random.randn(1, 512).astype('float32')
        cur_vec = torch.from_numpy(vecs_to_inject[0]).unsqueeze(0).to("cuda")
        res, latent = net(cur_vec, input_code=True, return_latents=True, resize=False, truncation=0.7)
        res_image = tensor2im(res[0])
    return res_image, latent
```
where `net` is a `pSp` object.

**Comment by tianjialiang12** [comment]
> I believe the issue is centered around what you're doing in the postprocessing on `src_img`. Try doing the following instead:
> 
> ```python
> src_img, _ = net.decodr([src_latent], input_is_latent=True, randomize_noise=False)
> image = tensor2im(src_img)
> ```
> 
> Also, I am also not sure without trying, but you may not want to perform `unsqueeze(0)` before passing the latent to the decoder so you should verify that this is indeed correct.
> If it helps, here is a function that generates random images using StyleGAN:
> 
> ```python
> def get_random_image(net):
>     with torch.no_grad():
>         vecs_to_inject = np.random.randn(1, 512).astype('float32')
>         cur_vec = torch.from_numpy(vecs_to_inject[0]).unsqueeze(0).to("cuda")
>         res, latent = net(cur_vec, input_code=True, return_latents=True, resize=False, truncation=0.7)
>         res_image = tensor2im(res[0])
>     return res_image, latent
> ```
> 
> where `net` is a `pSp` object.



Thank you very much for your answer. Now my project can work normally. By the way, this is the code.
`src_img, _ = net.decoder([src_latent], input_is_latent=True, randomize_noise=False)
img = tensor2im(src_img[0])
plt.axis('off')
plt.imshow(img)
plt.show()`
